javalin/src/main/java/io/javalin/http/servlet/ExceptionMapper.kt

```diff
@@ -40,11 +41,10 @@ class ExceptionMapper(val cfg: JavalinConfig) {
 
     internal fun handleUnexpectedThrowable(res: HttpServletResponse, throwable: Throwable): Nothing? {
         val unwrapped = (throwable as? CompletionException)?.cause ?: throwable
-        if (JettyUtil.isClientAbortException(unwrapped) || JettyUtil.isJettyTimeoutException(unwrapped)) {
-            JavalinLogger.debug("Client aborted or timed out", throwable)
-            return null // jetty aborts and timeouts happen when clients disconnect, they are not actually unexpected
-        }
         res.status = HttpStatus.INTERNAL_SERVER_ERROR.code
+        if (JettyUtil.isExpectedException(unwrapped)) {
+            return null.also { JettyUtil.logExpectedException(unwrapped) }
+        }
         JavalinLogger.error("Exception occurred while servicing http-request", throwable)
         return null
```

Comment:
It looks kinda strange 😅 What about sth like

when {
   JettyUtil.isExpectedException(unwrapped) -> JettyUtil.logExpectedException(unwrapped)
   else -> JavalinLogger.error("Exception occurred while servicing http-request", throwable)
}
return null

Reply:
Thanks, I was too focused on the if 😅